src/main/java/com/sap/oss/phosphor/fosstars/data/github/SecuregoDataProvider.java

```diff
@@ -0,0 +1,263 @@
+          }
+          if (runsWithSelectedRules(githubAction)) {
+            usesSecuregoWithSelectedRules = USES_SECUREGO_WITH_RULES.value(true);
+            break;
+          }
+        }
+      }
+    }
+
+    return ValueHashSet.from(runsSecurego, usesSecuregoWithSelectedRules, usesSecuregoScanChecks);
+  }
+
+  /**
+   * Checks if the Github action uses Securego sacn with rules.
+   *
+   * @param githubAction A config of the action
+   * @return True if the Github Action has config to run Securego with rules.
+   */
+  private boolean runsWithSelectedRules(Map<?, ?> githubAction) {
+    return Optional.ofNullable(githubAction.get("jobs"))
+        .filter(Map.class::isInstance)
+        .map(Map.class::cast)
+        .map(jobs -> jobs.values())
+        .filter(Iterable.class::isInstance)
+        .map(Iterable.class::cast)
+        .map(SecuregoDataProvider::checkAllJobs)
+        .get();
+  }
+
+  /**
+   * Checks if any step in a collection of jobs triggers a Securego scan with rules.
+   *
+   * @param jobs The collection of jobs from GitHub action.
+   * @return True if a step triggers a Securego scan with rules, false otherwise.
+   */
+  private static boolean checkAllJobs(Iterable<?> jobs) {
+    return IteratorUtils.toList(jobs.iterator()).stream()
+        .filter(Map.class::isInstance)
+        .map(Map.class::cast)
+        .map(job -> job.get("steps"))
+        .filter(Iterable.class::isInstance)
+        .map(Iterable.class::cast)
+        .map(SecuregoDataProvider::checkAllSteps)
+        .findAny()
+        .get();
+  }
+
+  /**
+   * Checks if a collection of steps from a GitHub action contains a step that triggers a Securego
+   * scan with rules.
+   *
+   * @param steps The steps to be checked.
+   * @return True if the steps contain a step that triggers a Securego scan with rules, false
+   *     otherwise.
+   */
+  private static boolean checkAllSteps(Iterable<?> steps) {
+    AtomicBoolean usesWithRules = new AtomicBoolean(false);
+    IteratorUtils.toList(steps.iterator()).stream()
+        .filter(Map.class::isInstance)
+        .map(Map.class::cast)
+        .forEach(
+            step -> {
+              if (securegoConfigExists(step.get("uses"))) {
+                usesWithRules.set(hasArgsWithRules(step.get("with")));
+              } else if (securegoConfigExists(step.get("run"))) {
+                usesWithRules.set(runsSecuregoWithRules(step.get("run")));
+              }
+            });
+    return usesWithRules.get();
+  }
+
+  /**
+   * Checks if a step contains config to run Securego.
+   *
+   * @param step The step to be checked.
+   * @return True if the step contains config to run Securego scan, false otherwise.
+   */
+  private static boolean securegoConfigExists(Object step) {
+    return Optional.ofNullable(step)
+        .filter(String.class::isInstance)
+        .map(String.class::cast)
+        .filter(run -> RUN_STEP_SECUREGO_REGEX_PATTERN.matcher(run).matches())
+        .isPresent();
```

Comment:
I think this should be 
```suggestion
        .map(run -> RUN_STEP_SECUREGO_REGEX_PATTERN.matcher(run).matches())
        .orElse(false);
```